### PR TITLE
silent renew: add option for silentRedirectUri in BrowserAuthClientConfig

### PR DESCRIPTION
**Pull request by aruniverse**
Fix silent sign in and renew. 

Need to set the `silent_redirect_uri` for the UserManager for this to work. See: https://gist.github.com/davidamidon/24c8a6980e116e62f781be4d6239d10d#signinsilent

Users can currently do this themselves via BrowserAuthorizationClient.setAdvancedSettings, but this is something we should do by default since we set [automatic silent renew to true](https://github1s.com/imodeljs/imodeljs/blob/HEAD/clients/frontend-authorization/src/oidc/browser/BrowserAuthorizationClient.ts#L102-L103) which gives the impression this works out of the box.

### CODE REVIEW

File: common/changes/@bentley/frontend-authorization-client/fix-silentRenew_2021-08-09-16-01.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@bentley/frontend-authorization-client",
+      "comment": "Add silrent redirect uri to BrowserAuthClientConfig. If not provided default to redirectUri",
```
Comment (aruniverse) [inline common/changes/@bentley/frontend-authorization-client/fix-silentRenew_2021-08-09-16-01.json:5]:
```suggestion
      "comment": "Add silent redirect uri to BrowserAuthClientConfig. If not provided default to redirectUri",
```

**Comment by aruniverse** [comment]
@Mergifyio backport release/2.19.x

**Review by calebmshafer — APPROVED** [review]
LGTM, you'll need to run `rush extract-api` on this.

Also, @evelynpreslar-bentley was the last to touch this code so should probably take a look too.

**Comment by evelynpreslar-bentley** [comment]
Silent renewal shouldn't be blocked by an undefined silent_redirect_uri argument, [because oidc-client-js internally falls back on redirect_uri for silent signins already](https://github.com/IdentityModel/oidc-client-js/blob/de8e7dea93c0ec8932c1975de23728cd9bc22198/src/UserManager.js#L250).

Do we have an example where silent token renewals are failing with the current configurations?

**Comment by aruniverse** [comment]
> Do we have an example where silent token renewals are failing with the current configurations?

We were having trouble in the iTwinDemo portal with the silent renew/refresh working until i manually set this myself. I have an open pr up there for this: https://github.com/iTwin/demo-portal/pull/59. DM me and i can send you a link to the actual site if you wanted to check it out

**Comment by aruniverse** [comment]
> LGTM, you'll need to run `rush extract-api` on this.

First time running this, seeing a lot of irrelevant changes wrt to my changes here. Is that because of this was done on master? Should I just run the extract-api on the pr created by the backport?

**Comment by calebmshafer** [comment]
@aruniverse you'll most likely need to do a merge with `master` and then run `rush clean && rush purge && rush install && rush build` and then run extract-api in the package you changed.

Or maybe try running `npm run extract-api` in the folder of the one you can't and don't go through all the other steps if it works :)

**Comment by calebmshafer** [comment]
@Mergifyio update

**Comment by calebmshafer** [comment]
@evelynpreslar-bentley @aruniverse any other concerns here or should we move forward?

**Comment by aruniverse** [comment]
> @evelynpreslar-bentley @aruniverse any other concerns here or should we move forward?

Looks like in the oidc-client library they do what I do by default. Started re-investigating what's going on today, need to continue tomorrow.

File: clients/frontend-authorization/src/oidc/browser/BrowserAuthorizationClient.ts [hunk 2]
```diff
@@ -100,6 +102,7 @@ export class BrowserAuthorizationClient extends BrowserAuthorizationBase<Browser
       post_logout_redirect_uri: basicSettings.postSignoutRedirectUri, // eslint-disable-line @typescript-eslint/naming-convention
       response_type: basicSettings.responseType, // eslint-disable-line @typescript-eslint/naming-convention
       automaticSilentRenew: true,
+      silent_redirect_uri: basicSettings.silentRedirectUri ?? basicSettings.redirectUri, // eslint-disable-line @typescript-eslint/naming-convention
```
Comment (aruniverse) [inline clients/frontend-authorization/src/oidc/browser/BrowserAuthorizationClient.ts:105]: @evelynpreslar-bentley thoughts? 

```suggestion
      silent_redirect_uri: basicSettings.silentRedirectUri, // eslint-disable-line @typescript-eslint/naming-convention
```

Comment (evelynpreslar-bentley) [reply clients/frontend-authorization/src/oidc/browser/BrowserAuthorizationClient.ts:105]: This is a fine change.
Simply enabling this option might be useful outside of setting it via advanced settings at some point.

**Comment by calebmshafer** [comment]
@Mergifyio update

**Comment by calebmshafer** [comment]
@Mergifyio update

**Comment by aruniverse** [comment]
@Mergifyio update

**Pull request merged**